src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs

```diff
@@ -186,7 +186,9 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
                             vsb.Append(input.AsSpan(prevat, match.Index - prevat));
 
                         prevat = match.Index + match.Length;
-                        vsb.Append(evaluator(match) ?? "");
+                        string result = evaluator(match);
+                        if (!String.IsNullOrEmpty(result))
```

Comment:
Nit: String => string

Reply:
oops. My last team had the opposite coding style..